src/widgets/models/CategoryModel.js

```diff
@@ -1,33 +1,45 @@
 import { executeSQL } from '../../api';
-import { filtersToSQL, viewportToSQL } from '../../api/FilterQueryBuilder';
+import { filtersToSQL } from '../../api/FilterQueryBuilder';
+import { applyFilter } from '../../api/Filter';
+import { groupValuesByColumn } from '../operations/groupby';
 
 export const getCategories = async (props) => {
-  const { data, credentials, column, operation, filters, viewport, opts } = props;
+  const { data, credentials, column, operation, filters, viewportFilter, viewportFeatures, type, opts } = props;
 
   const operationColumn = props.operationColumn || column;
 
   if (Array.isArray(data)) {
     throw new Error('Array is not a valid type to get categories');
   }
 
-  let query =
-    (viewport && `SELECT * FROM (${data})  as q WHERE ${viewportToSQL(viewport)}`) ||
-    data;
+  if (type === 'bq' && !viewportFilter) {
+    throw new Error('Category Widget error: BigQuery layers needs "viewportFilter" prop set to true.');
+  }
+
+  // It's an await because we probably will move this calculation need to a webworker
```

Comment:
Could you explain a bit more this comment?

Reply:
We were planning to move this to a web worker, for now it's not necessary, removing..